ocaml/xapi/test_vdi_cbt.ml

```diff
@@ -209,6 +234,52 @@ let test_allowed_operations_updated_when_necessary () =
   Client.Client.VDI.data_destroy ~rpc ~session_id ~self;
   assert_allowed_operations "does not contain `copy after VDI has been data-destroyed" (fun ops -> not @@ List.mem `copy ops)
 
+(* Confirm VDI.data_destroy changes requisite fields of VDI *)
+let test_vdi_after_data_destroy () =
+  try
+    let __context = Test_common.make_test_database () in
+    let sR,vDI = make_mock_server_infrastructure ~__context in
+    Db.VDI.set_type ~__context ~self:vDI ~value:`suspend;
+    let vM = Test_common.make_vm ~__context () in
+    let vBD = Test_common.make_vbd ~__context ~uuid:"VBD-1" ~vDI ~vM ~currently_attached:false () in
+
+    let check_vdi_is_snapshot_and_type ~vDI ~snapshot ~vdi_type ~managed =
+      OUnit.assert_equal ~msg:("VDI type should be set to " ^ (Record_util.vdi_type_to_string vdi_type))
+        (Db.VDI.get_type ~__context ~self:vDI) vdi_type;
+      OUnit.assert_equal ~msg:("VDI managed should be set to " ^ (if managed then "true" else "false"))
```

Comment:
These can be simplified using format strings: `Printf.sprintf ("VDI ... %b" managed)`, it's documented at https://caml.inria.fr/pub/docs/manual-ocaml/libref/Printf.html